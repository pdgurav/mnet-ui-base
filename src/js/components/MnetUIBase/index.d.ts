import * as React from 'react';
import { ThemeType } from '../../themes';
import { BackgroundType } from '../../utils';

export interface MnetUIBaseProps {
  background?: BackgroundType;
  containerTarget?: HTMLElement;
  cssVars?: boolean;
  dir?: 'rtl';
  full?: boolean | 'min';
  plain?: boolean;
  theme?: ThemeType;
  themeMode?: 'dark' | 'light';
  userAgent?: string;
}

export interface MnetUIBaseExtendedProps
  extends MnetUIBaseProps,
    Omit<JSX.IntrinsicElements['div'], 'dir'> {}

declare const MnetUIBase: React.FC<MnetUIBaseExtendedProps>;

export { MnetUIBase };
