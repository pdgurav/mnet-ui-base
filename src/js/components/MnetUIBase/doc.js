import { describe, PropTypes } from 'react-desc';

import { backgroundDoc } from '../../utils/prop-types';
import { getAvailableAtBadge } from '../../utils/mixins';

export const doc = MnetUIBase => {
  const DocumentedMnetUIBase = describe(MnetUIBase)
    .availableAt(getAvailableAtBadge('MnetUIBase', 'Utilities'))
    .description('The top level MnetUIBase container.')
    .usage(
      `import { MnetUIBase } from 'mnet-ui-base';
<MnetUIBase>...</MnetUIBase>`,
    )
    .intrinsicElement('div');

  DocumentedMnetUIBase.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description(
      'Layout direction for right to left contexts',
    ),
    full: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['min'])])
      .description(
        `Whether to take the whole viewport. 'min' indicates it should
        take at minimum the whole viewport, allowing the viewport to
        control scrolling.`,
      )
      .defaultValue(false),
    plain: PropTypes.bool
      .description(
        `Whether or not MnetUIBase should apply a global font-family, font-size,
        and line-height.`,
      )
      .defaultValue(false),
    cssVars: PropTypes.bool
      .description('Whether to expose the css variables.')
      .defaultValue(false),
    theme: PropTypes.object.description(
      'Custom styles for MnetUIBase app component.',
    ),
    themeMode: PropTypes.oneOf(['dark', 'light']).description(
      `Dark vs. light theme variation. Default is unspecified and left to
      theme.`,
    ),
    userAgent: PropTypes.string.description(
      `User agent used to detect the device width for setting the initial
      breakpoint.`,
    ),
    containerTarget: PropTypes.object.description(
      `The node where Drop and Layer containers are inserted. Defaults to
      document.body which is almost always the right choice. This is used
      for less common cases like rendering within an internal node (e.g.
      shadow root).`,
    ),
  };

  return DocumentedMnetUIBase;
};

export const themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined,
  },
};
