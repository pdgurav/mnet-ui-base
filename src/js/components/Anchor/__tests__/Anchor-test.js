import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import { MnetUIBase } from '../../MnetUIBase';
import { Anchor } from '..';

describe('Anchor', () => {
  afterEach(cleanup);

  test('renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with children', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor href="#">children</Anchor>
      </MnetUIBase>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('warns about invalid label render', () => {
    const warnSpy = jest.spyOn(console, 'warn');
    const component = render.create(
      <MnetUIBase>
        <Anchor href="#" label="Test">
          invalid
        </Anchor>
      </MnetUIBase>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(warnSpy).toHaveBeenCalledWith(
      'Anchor should not have children if icon or label is provided',
    );

    warnSpy.mockReset();
    warnSpy.mockRestore();
  });

  test('warns about invalid icon render', () => {
    const warnSpy = jest.spyOn(console, 'warn');
    const component = render.create(
      <MnetUIBase>
        <Anchor href="#" icon={<svg />}>
          invalid
        </Anchor>
      </MnetUIBase>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(warnSpy).toHaveBeenCalledWith(
      'Anchor should not have children if icon or label is provided',
    );

    warnSpy.mockReset();
    warnSpy.mockRestore();
  });

  test('primary renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor href="#" primary label="Test" />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('focus renders', () => {
    const { container, getByText } = render(
      <MnetUIBase>
        <Anchor href="#" label="Test" />
      </MnetUIBase>,
    );
    fireEvent.focus(getByText('Test'));
    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor disabled />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('icon label renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor icon={<svg />} label="Test" onClick={() => {}} />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('reverse icon label renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor reverse icon={<svg />} label="Test" onClick={() => {}} />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('size renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor size="xsmall" />
        <Anchor size="small" />
        <Anchor size="medium" />
        <Anchor size="large" />
        <Anchor size="xlarge" />
        <Anchor size="xxlarge" />
        <Anchor size="10px" />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('is clickable', () => {
    const onClick = jest.fn();
    const { container, getByText } = render(
      <MnetUIBase>
        <Anchor href="#" label="Test" onClick={onClick} />
      </MnetUIBase>,
    );
    const anchor = getByText('Test');
    fireEvent.click(anchor);
    expect(container.firstChild).toMatchSnapshot();
    expect(onClick).toBeCalled();
  });

  test('renders tag', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor href="#" label="Test" as="span" />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });

  test('weight renders', () => {
    const { container } = render(
      <MnetUIBase>
        <Anchor href="#" label="Normal" weight="normal" />
        <Anchor href="#" label="Bold" weight="bold" />
        <Anchor href="#" label="Bold" weight={500} />
      </MnetUIBase>,
    );
    expect(container).toMatchSnapshot();
  });
});
