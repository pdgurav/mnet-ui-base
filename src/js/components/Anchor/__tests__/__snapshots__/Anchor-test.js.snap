// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor disabled renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  opacity: 0.3;
  cursor: default;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      disabled=""
    />
  </div>
</div>
`;

exports[`Anchor focus renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  box-shadow: 0 0 2px 2px #B1C2FE;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 > circle,
.c1 > ellipse,
.c1 > line,
.c1 > path,
.c1 > polygon,
.c1 > polyline,
.c1 > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #B1C2FE;
}

.c1::-moz-focus-inner {
  border: 0;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    Test
  </a>
</div>
`;

exports[`Anchor icon label renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 4px;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
    >
      <span
        class="c2"
        style="display: inline-flex;"
      >
        <svg
          color="#7D4CDB"
        />
        <span
          class="c3"
        />
        Test
      </span>
    </a>
  </div>
</div>
`;

exports[`Anchor is clickable 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    Test
  </a>
</div>
`;

exports[`Anchor primary renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="#"
    >
      Test
    </a>
  </div>
</div>
`;

exports[`Anchor renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
    />
  </div>
</div>
`;

exports[`Anchor renders tag 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
      href="#"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Anchor renders with children 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    children
  </a>
</div>
`;

exports[`Anchor reverse icon label renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 4px;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
    >
      <span
        class="c2"
        style="display: inline-flex;"
      >
        Test
        <span
          class="c3"
        />
        <svg
          color="#7D4CDB"
        />
      </span>
    </a>
  </div>
</div>
`;

exports[`Anchor warns about invalid icon render 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 4px;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
    >
      Link
    </a>
  </div>
</div>
`;

exports[`Anchor size renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: 12px;
  line-height: 18px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  box-sizing: border-box;
  font-size: 14px;
  line-height: 20px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  box-sizing: border-box;
  font-size: 18px;
  line-height: 24px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4 {
  box-sizing: border-box;
  font-size: 22px;
  line-height: 28px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  box-sizing: border-box;
  font-size: 26px;
  line-height: 32px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  box-sizing: border-box;
  font-size: 34px;
  line-height: 40px;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7 {
  box-sizing: border-box;
  font-size: 10px;
  line-height: normal;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
    />
    <a
      class="c2"
    />
    <a
      class="c3"
    />
    <a
      class="c4"
    />
    <a
      class="c5"
    />
    <a
      class="c6"
    />
    <a
      class="c7"
    />
  </div>
</div>
`;

exports[`Anchor warns about invalid icon render 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  padding: 12px;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    <svg
      color="#E15151"
    />
  </a>
</div>
`;

exports[`Anchor warns about invalid label render 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="#"
  >
    Test
  </a>
</div>
`;

exports[`Anchor weight renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <a
      class="c1"
      href="#"
    >
      Normal
    </a>
    <a
      class="c2"
      href="#"
    >
      Bold
    </a>
    <a
      class="c3"
      href="#"
    >
      Bold
    </a>
  </div>
</div>
`;
